client/components/popover/index.jsx

```diff
@@ -1,160 +1,379 @@
+		}
+
+		this.setPosition();
+		this.bindClickoutHandler( domContainer );
+	}
+
+	getPositionClass( position = this.props.position ) {
+		return `popover-${ position.replace( /\s+/g, '-' ) }`;
+	}
+
+	/**
+	 * Computes the position of the Popover in function
+	 * of its main container and the target.
+	 *
+	 * @return {Object} reposition parameters
+	 */
+	computePosition() {
+		if ( ! this.props.isVisible ) {
+			return null;
+		}
+
+		const { domContainer, domContext } = this;
+		const { position } = this.props;
+
+		if ( ! domContainer || ! domContext ) {
+			this.debug( '[WARN] no DOM elements to work' );
+			return null;
+		}
+
+		const suggestedPosition = suggestPosition( position, domContainer, domContext );
+
+		this.debug( 'suggested position: %o', suggestedPosition );
+
+		const reposition = Object.assign(
+			{},
+			constrainLeft(
+				offset( suggestedPosition, domContainer, domContext ),
+				domContainer
+			),
+			{ positionClass: this.getPositionClass( suggestedPosition ) }
+		);
+
+		this.debug( 'updating reposition: ', reposition );
+
+		return reposition;
+	}
+
+	setPosition() {
+		const position = this.computePosition();
+		if ( ! position ) {
+			return null;
+		}
+
+		this.willReposition = null;
+		this.setState( position );
+	}
+
+	getStylePosition() {
+		const { left, top } = this.state;
+		return { left, top };
+	}
+
+	show() {
+		// take over of its own visibility
+		if ( this._timer ) {
+			clearTimeout( this._timer );
+			this._timer = null;
+		}
+
+		this._timer = setTimeout( () => {
+			this.setState( { show: true } );
+		}, this.props.openDelay );
+	}
+
+	hide() {
+		this.setState( { show: false } );
+		if ( this._timer ) {
+			clearTimeout( this._timer );
+			this._timer = null;
+		}
+	}
+
+	close() {
+		if ( ! this.props.isVisible ) {
+			this.debug( 'popover should be already closed' );
+			return null;
+		}
+
+		this.props.onClose();
+	}
+
+	render() {
+		const { show } = this.state;
+		const { context, className } = this.props;
+
+		if ( ! show ) {
+			this.debug( 'Popover is not visible.' );
+			return null;
+		}
+
+		if ( ! context ) {
+			this.debug( 'No `context` to tie the Popover' );
+			return null;
+		}
+
+		const classes = classNames(
+			className,
+			this.state.positionClass
+		);
+
+		this.debug( 'rendering ...' );
+
+		return (
+			<RootChild>
+				<div
+					style={ this.getStylePosition() }
+					className={ classes }
+					ref={ this.setDOMBehavior }
+				>
+					<div className="popover__arrow" />
+
+					<div className="popover__inner">
+						{ this.props.children }
+					</div>
+				</div>
+			</RootChild>
+		);
+	}
+}
+
+Popover.propTypes = {
+	className: PropTypes.string,
+	closeOnEsc: PropTypes.bool,
+	openDelay: PropTypes.number,
+	position: PropTypes.string,
+	id: PropTypes.string,
+
+	onClose: PropTypes.func.isRequired,
+	onShow: PropTypes.func,
+};
+
+Popover.defaultProps = {
+	className: 'popover__container',
+	closeOnEsc: true,
+	openDelay: 100,
```

Comment:
Sorry by the delay :-p

This parameter, renamed to `showDelay`, add debounce when the Popover is opened. It optimizes some edge cases when we have a lot of targets all together.

Although it's outside my area I think also it improves the UX.
https://github.com/Automattic/wp-calypso/pull/7217#issuecomment-237300078

Anyway this parameter is optional and the default options isn't add the delay.